### PR TITLE
More ui fixes

### PR DESCRIPTION
**Pull request by fmatthew5876**
Fixes more from #1443

### CODE REVIEW

**Comment by Ghabry** [comment]
Quick Note: A misalignment of ~2 pixels on y could be also because of the built-in font used because our naive pixel font renderer doesn't use a baseline and not a problem with the scene.

There are two fonts. you can switch between them via the "system" graphic setting in the editor.

Best way to test is with a glyph which takes the whole 12 pixels of height.
Here is the pixel font: https://github.com/EasyRPG/Player/tree/master/resources/shinonome

**Comment by fmatthew5876** [comment]
We seem to hard code a y offset of 2 in many window classes

**Comment by Ghabry** [comment]
k, then I need to check this. Don't know which ones but for some windows I measured the font position via screenshots and paint ^^

**Comment by fmatthew5876** [comment]
Take a look at Window_Help. It's almost exactly like this. 

Also my last PR for status had y = 2

**Comment by fmatthew5876** [comment]
I added animation for the file menu slots but there is a problem. RPG_RT keeps a small blue rectangle above and below the 3 displayed file menu boxes.

Whats the best way to preserve this behavior in Player? Do we have a mechanism to define a clipping rectangle so that as the file box animates up/down it wont cover the border? 

If not, I guess the other way would be to actually render 2 blue rectangles ontop of the windows?

**Comment by fmatthew5876** [comment]
How can I draw single color `Sprite()`?

I've tried this but I don't see anything. What am I missing?

```
auto* border = new Sprite();
border->SetSrcRect(Rect(0, 32, SCREEN_TARGET_WIDTH, 8));                      
border->SetVisible(true);
border->SetZ(Priority_Window + 10);
border->SetBitmap(Bitmap::Create(0, 32, SCREEN_TARGET_WIDTH, 8));
border->GetBitmap()->Fill(Color(255,255,255,255));
```

**Comment by Ghabry** [comment]
SetSrcRect is not needed and GetBitmap is const, you must draw on the Bitmap before you give it to the sprite.

**Comment by Ghabry** [comment]
You can use "SpriteRef". SetSrcRect is not needed, same for Update. You invoke the wrong Bitmap::Create constructor, guess this invokes width, height, transparent, bpp. Guess you want to invoke width, height, Color

**Comment by Ghabry** [comment]
and the system background is Cache::System()->GetBackgroundColor()

**Comment by fmatthew5876** [comment]
I added the border sprites and now the animation matches `RPG_RT`.

File: src/scene_file.cpp [hunk 1]
```diff
@@ -109,23 +124,31 @@ void Scene_File::Start() {
...
+
 	index = latest_slot;
 
 	Refresh();
 	Update();
 }
 
 void Scene_File::Refresh() {
-	for (unsigned int i = 0; (size_t) i < file_windows.size(); i++) {
+	for (int i = 0; i < int(file_windows.size()); i++) {
```
Comment (carstene1ns) [inline src/scene_file.cpp]: this looks really ugly.

Comment (fmatthew5876) [reply src/scene_file.cpp]: Changed it to a cast. The loop variable should be signed as it is compared to signed values inside the loop.

**Comment by fmatthew5876** [comment]
It's actually animating the windows super fast. The delta in space is x15 but the delta in time is the same.

Using PageUp / PageDown also scrolls 3 times faster

I like this better than an instant transition. All motions up, down, pageup, pagedown all have the same speed and delay.

**Pull request merged**